python3/packages/observer.py

```diff
@@ -288,12 +289,13 @@ def instrument_function(wrapped, _, args, kwargs):
                     # class or classmethod
                     aspan.set_attribute("xs.span.args.str", str(args))
                     aspan.set_attribute("xs.span.kwargs.str", str(kwargs))
-                else:
-                    # function, staticmethod or instancemethod
+                elif type(wrapped)==types.FunctionType or type(wrapped)==types.MethodType:
                     bound_args = inspect.signature(wrapped).bind(*args, **kwargs)
                     bound_args.apply_defaults()
                     for k, v in bound_args.arguments.items():
                         aspan.set_attribute(f"xs.span.arg.{k}", str(v))
+                else: #eg. class 'PartialCallableObjectProxy'
+                    pass
```

Comment:
While we should not trust AI tools alone, the https://sourcery.ai plugin also notifies me of the superfluous pass:
```suggestion
```